### PR TITLE
Raise exception if the user passes in unused keywords to Client

### PR DESCRIPTION
**Pull request by detroyejr**
Exceptions are raised using the simple approach [suggested by mrocklin](https://github.com/dask/distributed/issues/3014#issuecomment-538349320).

It seems that some tests like [test_client_with_name](https://github.com/dask/distributed/blob/670f2e84a7607b685cbc2f8fbdbfb29aa8267c84/distributed/tests/test_client.py#L5014) fail since they pass in keywords like `silence_logs`.

In this case, are we in favor of working around these tests until they pass or change where these extra arguments are placed? What are best practices here?

#3014

### CODE REVIEW

**Comment by detroyejr** [comment]
Some of these tests I expected to fail. Not sure about all the `gen_cluster` failures. Will do some digging.

File: distributed/client.py [hunk 1]
```diff
@@ -643,6 +643,17 @@ def __init__(
             if address:
                 logger.info("Config value `scheduler-address` found: %s", address)
 
+        if address is not None and kwargs:
+            unused_arg = [
+                arg
+                for arg in kwargs
+                if arg not in ["silence_logs"] and kwargs.get(arg) is not None
+            ]
```
Comment (mrocklin) [inline distributed/client.py]: I'm curious, why did you have to include these exceptions?  

My guess is that you're working around test failures somewhere in the test suite.  It's entirely possible though that those tests are wrong.

File: distributed/deploy/tests/test_local.py [hunk 2]
```diff
@@ -219,6 +219,27 @@ def test_Client_kwargs(loop):
...
+def test_Client_unused_kwargs_with_address(loop):
+    with pytest.raises(Exception) as argexcept:
+        c = Client(
+            "127.0.0.1:8786", n_workers=2, dashboard_port=8000, silence_logs=None
+        )
+    assert (
+        str(argexcept.value)
+        == "Unexpected keyword arguments: ['dashboard_port', 'n_workers']"
+    )
+
```
Comment (mrocklin) [inline distributed/deploy/tests/test_local.py]: Thanks for this test!

File: distributed/client.py (hunk 1 above)
Comment (detroyejr) [reply distributed/client.py]: Yes, this is just to get the current test suite to pass. Some tests like `test_client_repr_closed` include arguments that cause the test to fail:

```
c = yield Client(s.address, asynchronous=True, dashboard_address=None)
```

While `dashboard_address` doesn't have a value, kwargs is no longer None/False so it's going to raise an exception. [Here's the example in travis logs.](https://travis-ci.org/dask/distributed/jobs/593684024#L3253)

I think there are only 2 or 3 tests that would need to be changed. I can do that if that's the better than working around them.

Comment (mrocklin) [reply distributed/client.py]: > I think there are only 2 or 3 tests that would need to be changed. I can do that if that's the better than working around them.

Yes, that would be ideal if you're willing.  Thank you for fixing more bugs :)

**Comment by mrocklin** [comment]
Merging this in. Thanks @detroyejr !

Also.  I notice that this is your first code contribution to this repository.  Welcome!

**Pull request merged**